src/main/java/org/openlmis/fulfillment/extension/point/DefaultOrderCodeGenerator.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *  
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+package org.openlmis.fulfillment.extension.point;
+
+import org.openlmis.fulfillment.domain.Order;
+import org.openlmis.fulfillment.domain.OrderNumberConfiguration;
+import org.openlmis.fulfillment.repository.OrderNumberConfigurationRepository;
+import org.openlmis.fulfillment.service.referencedata.ProgramDto;
+import org.openlmis.fulfillment.service.referencedata.ProgramReferenceDataService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * A default implementation of OrderCodeGenerator extension point.
+ * @see OrderCodeGenerator
+ */
+@Component("DefaultOrderCodeGenerator")
+public class DefaultOrderCodeGenerator implements OrderCodeGenerator {
+
+  @Autowired
+  private OrderNumberConfigurationRepository orderNumberConfigurationRepository;
+
+  @Autowired
+  private ProgramReferenceDataService programReferenceDataService;
+
+  /**
+   * Generates unique number for given order.
+   */
+  public String generate(Order order) {
+    OrderNumberConfiguration orderNumberConfiguration =
```

Comment:
I didn't expect to see this as this implementation is the entire usage of the Order Number Configuration.  I thought that if we defined an extension point, that if the configuration was:

<some configurable prefixes>OrderUUID<some configurable sufixes>

That we'd define the extension point just for the middle bit:  Order UUID.

The reason this is important is that we do have on the roadmap to be able to configure an order number through an administration screen (UI).  If we have this extension point that would potentially do anything it likes with the entire order number configuration, I'm thinking that we're already setting ourselves to do duplicate work.  What if we just put an extension point for the OrderUUID.  In that way the configuration would apply, however the choice of a UUID could be extended to base36 or any other scheme that's inside the configurable prefix and suffix.

Reply:
Since we are generating only the inner part, is it alright to change extension point's name to OrderNumberGenerator?